### PR TITLE
Plots.jl recipes to plot GPs  

### LINKED ISSUE #25: Plotting Recipes
It's useful to be able to produce nice pretty plots of GPs in 1 dimension. Currently, this is a bit of a pain because no plotting recipes have been implemented. If we implemented some, it would significantly simplify a _lot_ of the plotting that we do.

(If you're interested in making this happen, I'm happy to discuss)

### PR DESCRIPTION
**Pull request by nathbo**
Fixes #25

Plotting now supports the following signatures:
``` julia
plot(gp::Stheno.AbstractGP, x::Array)
plot(gp::Stheno.AbstractGP, x::AbstractRange)
plot(gp::Stheno.AbstractGP, xmin::Number, xmax::Number)
plot(gp::Stheno.FiniteGP)
```

Additionally `plot` accepts a `samples` keyword argument and it is possible to define a seriestype, mostly relevant to decide between `:line` and `:scatter` plots.
``` julia
plot(gp::Stheno.FiniteGP; samples=0, sample_seriestype=:line)
```

Please let me know what you think!

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -10,6 +10,7 @@ FillArrays = "1a297f60-69ca-5386-bcde-b61e274b549b"
 IRTools = "7869d1d1-7146-5819-86e3-90919afe41df"
 LinearAlgebra = "37e2e46d-f89d-539d-b4ee-838fcccc9c8e"
 MacroTools = "1914dd2f-81c6-5fcd-8719-6d5c9610ff09"
+Plots = "91a5bcdd-55d7-5caf-9e0b-520d859cae80"
```
Comment (willtebbutt) [inline Project.toml]: Please swap this out for [RecipesBase.jl](https://github.com/JuliaPlots/RecipesBase.jl) - it's a dependency-free package that provides only the stuff that you need for defining recipes, without any of the plotting baggage

**Comment by willtebbutt** [comment]
Your gists looks great btw. Would you mind adding the plotting code in them to the bottom of the examples on the readme? I previously didn't have the plotting code there because it was too verbose, but it will look really nice now :)

**Comment by nathbo** [comment]
Thank you for your positive feedback! I will change the dependency and include the plotting in the readme. Since I did so in the gist I will remove the explicit computation of the marginals, means, std-devs and samples, but if you think that these calls would be helpful for others it could also be reasonable to leave them.

**Comment by willtebbutt** [comment]
Great. I would be pro- leaving those calls there for pedagogical purposes.

File: README.md [hunk 2]
```diff
@@ -128,7 +171,7 @@ The plan is again not to support the combination of GPs and Deep Learning explic
 
 ## Things that are definitely up for grabs
 Obviously, improvements to code documentation are always welcome, and if you want to write some more unit / integration tests, please feel free. In terms of larger items that require some attention, here are some thoughts:
-- Plotting recipes: there is currently a lot of _highly_ repetitive code for plotting the posterior distribution over 1D GPs. This needn't be the case, and it would be a (presumably) simple job for someone who knows what they're doing with the [Plots.jl](https://github.com/JuliaPlots/Plots.jl) recipes to make most of that code disappear. 
+- Plotting recipes: there is currently a lot of _highly_ repetitive code for plotting the posterior distribution over 1D GPs. This needn't be the case, and it would be a (presumably) simple job for someone who knows what they're doing with the [Plots.jl](https://github.com/JuliaPlots/Plots.jl) recipes to make most of that code disappear.
```
Comment (willtebbutt) [inline README.md]: This can probably go haha

File: README.md [hunk 3]
```diff
@@ -58,6 +58,39 @@ ms3 = marginals(f₃′(Xp));
 μf₁′, σf₁′ = mean.(ms1), std.(ms1);
 μf₂′, σf₂′ = mean.(ms2), std.(ms2);
 μf₃′, σf₃′ = mean.(ms3), std.(ms3);
+
+# Instantiate plot and chose backend
+plotly();
+posterior_plot = plot();
+
+# Plot posteriors
+plot!(posterior_plot, f₁′(Xp); samples=S,
```
Comment (willtebbutt) [inline README.md]: Nit picky thing: these can probably go on one line as they're short enough.

File: README.md [hunk 4]
```diff
@@ -108,6 +141,41 @@ ms3 = marginals(y₂′(Xp));
 μf′, σf′ = mean.(ms1), std.(ms1);
 μy₁′, σy₁′ = mean.(ms2), std.(ms2);
 μy₂′, σy₂′ = mean.(ms3), std.(ms3);
+
+# Instantiate plot and chose backend
+plotly();
+posterior_plot = plot();
+
+# Plot posteriors
+plot!(posterior_plot, y₁′(Xp);
```
Comment (willtebbutt) [inline README.md]: Same here. They're just about under the 92 char limit, so one line is fine :)

**Comment by nathbo** [comment]
I included all the mentioned changes in the latest commit, I hope I did not forget anything. 

One other thing: I see a "All checks have failed" notification, but I don't really know anything about travis-ci. Should I change anything else in order to pass the tests?

File: README.md [hunk 5]
```diff
@@ -108,6 +135,33 @@ ms3 = marginals(y₂′(Xp));
 μf′, σf′ = mean.(ms1), std.(ms1);
 μy₁′, σy₁′ = mean.(ms2), std.(ms2);
 μy₂′, σy₂′ = mean.(ms3), std.(ms3);
+
+# Instantiate plot and chose backend
+plotly();
+posterior_plot = plot();
+
+# Plot posteriors
+plot!(posterior_plot, y₁′(Xp); samples=S, sample_seriestype=:scatter, color=:red, label="")
```
Comment (willtebbutt) [inline README.md]: Add semi-colons here to suppress plotting intermediates?

**Comment by willtebbutt** [comment]
Almost good to go.

Regarding the failing tests, somehow the `Manifest.toml` has worked its way back into the repo. This is a bad thing and might be the cause of the failures. It's probably a mistake that I made in a previous commit as it's on master as well, but could you try removing it and seeing if the tests pass?

File: README.md (hunk 5 above)
Comment (nathbo) [reply README.md]: My bad, you are absolutely right :)

**Comment by willtebbutt** [comment]
Removing the Manifest seems to have sorted out the load issues. Thanks for turning this around so quickly, will merge once tests have passed + docs built.

**Comment by willtebbutt** [comment]
@nathbo are you on the Julia slack or anything?

**Pull request merged**